config/jobs/kubernetes/cloud-provider-gcp/cloud-provider-gcp-periodics.yaml

```diff
@@ -324,7 +319,10 @@ periodics:
           echo "TEST_PACKAGE_VERSION - Falling back to v1.25.0";
         fi;
         cd $REPO_ROOT;
+        KUBE_VERSION=${KUBE_VERSION:-master};
+        KUBE_SOURCE=_src/k8s.io/kubernetes;
         sed -i 's/^\s*k8s.io.*//g' go.mod;
+        git clone --depth 1 --branch ${KUBE_VERSION} https://github.com/kubernetes/kubernetes ${KUBE_SOURCE};
```

Comment:
uh ... the cloning in prow is pretty flexible, I would prefer we not wind up with a half-broken cloning implementation ....

there is also excessive logic inline in this job, please can we move this into a script in the repo under test?
this makes it easier to iterate on fixes and removes load from test-infra approvers from reviewing every detail of CI jobs

Reply:
it's a circular dependency, the final state is as you say, see PR here
https://github.com/kubernetes/cloud-provider-gcp/pull/462

but that repo is going under a big refactor and I want to get signal first